### PR TITLE
Fix some typos in comments

### PR DESCRIPTION
**Pull request by mauke**

### CODE REVIEW

**Review by leonerd — APPROVED** [review]
LGTM

**Comment by demerphq** [comment]
Looks like you committed with an unregistered email address. Either rebase it with an author override to your canonical author name or run updateAuthors.pl

**Comment by mauke** [comment]
> Looks like you committed with an unregistered email address. Either rebase it with an author override to your canonical author name or run updateAuthors.pl

How do I do that in the Github browser interface? (Also, I've committed stuff from `l.mai AT web.de` before. Why is this not automatic?)

**Comment by demerphq** [comment]
> How do I do that in the Github browser interface?

No idea, I suspect you can't.

> Also, I've committed stuff from l.mai AT web.de before. Why is this not automatic?)

Because it is not just a matter of your email address (we have different committers with the same email address in our records). 

If you had run a `make test_porting` (or make test) before you pushed you would have seen this:

```
#   Failed test 'Is mailmap_file '.mailmap' up to date?'
#   at Porting/updateAUTHORS.pl line 126.
#   File '.mailmap' changes:
#     would add: Lukas Mai <l.mai@***.de> mauke <l.mai@***.de>
# 
# Files need updating! You probably just need to run
# 
#    Porting/updateAUTHORS.pl
# 
# and commit the results.
# Looks like you failed 1 test of 5.
```

Notice that your *name* is "mauke" in the latter, and Lukas Mai in the former canonical entry in AUTHORS.  We map all aliases to the canonical name in AUTHORS as part of a general effort of ensuring that AUTHOR is automagically maintained.  We can't do it automatically because that would mean automatically committing changes to the repo. 

If for some reason it is too much trouble for you to address this yourself then I can do it for you, but please tell me what you prefer, would you like me to add your "mauke" email to the .mailmap or would like me to rebase your commit to use your canonical name?

Yves

**Comment by mauke** [comment]
> > How do I do that in the Github browser interface?
> 
> No idea, I suspect you can't.
> 
> > Also, I've committed stuff from l.mai AT web.de before. Why is this not automatic?)
> 
> Because it is not just a matter of your email address (we have different committers with the same email address in our records).

Ah. That's unfortunate. Thanks for the explanation.

> If you had run a `make test_porting` (or make test) before you pushed you would have seen this:
> 
> ```
> #   Failed test 'Is mailmap_file '.mailmap' up to date?'
> #   at Porting/updateAUTHORS.pl line 126.
> #   File '.mailmap' changes:
> #     would add: Lukas Mai <l.mai@***.de> mauke <l.mai@***.de>
> # 
> # Files need updating! You probably just need to run
> # 
> #    Porting/updateAUTHORS.pl
> # 
> # and commit the results.
> # Looks like you failed 1 test of 5.
> ```

I didn't "push" per se. I was browsing pp_hot.c in the perl repo on Github to figure out some pp_multideref details when I noticed a few comments with mistakes. I clicked the "edit" button, made my changes, and saved. Github then automatically offered the option to create a pull request.

That said, the `.github/CONTRIBUTING.pod` file is pointless. It is a symlink, and Github doesn't support symlinks in the web UI, so it just looks like nothing (or rather, like a text file containing `../pod/perlhack.pod`, which isn't helpful): https://github.com/Perl/perl5/blob/blead/.github/CONTRIBUTING.pod

It might be better to make `.github/CONTRIBUTING.pod` a copy of `perlhack.pod` and add a porting test to make sure they stay in sync. It's a bit dumb, but at least it'll look better on Github.

I have now cloned the repository locally and added my mail alias, but `t/porting/cmp_version.t` is still failing:

```
...
ok 573 - in cpan/Encode 2 files were modified and a version was bumped
# No versions bumped in dist/Devel-PPPort but 2 files were modified
# dist/Devel-PPPort/module2.c
# dist/Devel-PPPort/module3.c
not ok 574 - in dist/Devel-PPPort 2 files were modified but no versions were bumped
# not ok 574 - in dist/Devel-PPPort 2 files were modified but no versions were bumped

ok 575 - in dist/IO 2 files were modified and a version was bumped
ok 576 - in dist/threads 1 file was modified and a version was bumped
ok 577 - in ext/DynaLoader 1 file was modified and a version was bumped
ok 578 - in ext/File-Glob 2 files were modified and a version was bumped
ok 579 - in ext/SDBM_File 10 files were modified and a version was bumped
# No versions bumped in ext/Win32CORE but 1 file was modified
# ext/Win32CORE/Win32CORE.c
not ok 580 - in ext/Win32CORE 1 file was modified but no versions were bumped
# not ok 580 - in ext/Win32CORE 1 file was modified but no versions were bumped
```

i don't know what's up with that; I never touched those files.

**Comment by demerphq** [comment]
> i don't know what's up with that; I never touched those files.

Are you on win32 or something like that? Any context info you can provide would be helpful.

**Comment by mauke** [comment]
> Are you on win32 or something like that? Any context info you can provide would be helpful.

No, this is a pretty boring Ubuntu Linux machine.

I am on commit e792c028462d5d3771e15dff0293056e2eab3af6.
```
$ git status
On branch blead
Your branch is up to date with 'origin/blead'.

nothing to commit, working tree clean
```
`make test_porting` reports:
```
...
porting/cmp_version.t ..... warning: exhaustive rename detection was skipped due to too many files.
warning: you may want to set your diff.renameLimit variable to at least 3669 and retry the command.
porting/cmp_version.t ..... 557/580 # not ok 574 - in dist/Devel-PPPort 2 files were modified but no versions were bumped

# not ok 580 - in ext/Win32CORE 1 file was modified but no versions were bumped

porting/cmp_version.t ..... Failed 2/580 subtests 
...
```

**Comment by demerphq** [comment]
> It might be better to make .github/CONTRIBUTING.pod a copy of perlhack.pod and add a porting test to make sure they stay in sync. It's a bit dumb, but at least it'll look better on Github.

I think @bram-perl had some opinions on that. There was some discussion on it in the past. What you say makes sense to me.

**Comment by demerphq** [comment]
> make test_porting reports:

That is really weird. Same thing for me (albeit in parallel mode) shows no changes.

It would be interesting to know if 

```
git clean -dfx
./Configure -Dusethreads -Doptimize=-g -d -Dusedevel -DDEBUGGING
make -j16 test_prep
TEST_JOBS=16 make -j16 test_porting
```

fixes it.

**Comment by mauke** [comment]
No, that fails the author test again (spuriously, I think):
```
# There are uncommitted changes in the working directory
# ?? l1_char_class_tab.h-new                  
```
But also:
```
porting/cmp_version.t   (Wstat: 0 Tests: 580 Failed: 2)
  Failed tests:  574, 580
```

**Comment by demerphq** [comment]
> No, that fails the author test again (spuriously, I think):

Curious, when I check out your  36c954871125d4fe48b776f559b6028be6ad5dc6
it passes test_porting just fine.

**Comment by demerphq** [comment]
Can you repush https://github.com/Perl/perl5/commit/f9ff8d072174a6af7a5dfec73d8dbe8e59909ba5 as a branch so i can get a copy of it to see what is up?

**Comment by mauke** [comment]
Haha, I've got it. When you clone ("fork") a repository on Github and later "synchronize" it with the original, it does not pull in any newer tags. So even though my local clone is on the right commit in the "blead" branch, it lacks all the tags from v5.28 or so upwards. And the `cmp_version.t` test crucially relies on the presence of tags.

TL;DR: The version test thought I was on tag `v5.27.5`.

**Comment by demerphq** [comment]
Thanks, merged. Sorry for the inconvenience.

**Comment by demerphq** [comment]
@mauke, I created https://github.com/Perl/perl5/pull/20926 in follow up to the issues you had with this PR today.
